.github/workflows/go.yml

```diff
@@ -0,0 +1,46 @@
+
+name: Go
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Build
+      run: go build -v ./...
+  
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      
+    - name: Test
+      run: go test -v ./...
+        
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          ref: ${{ github.head_ref }}
+      - name: go fmt
+        run: go fmt ./...
+      - name: Check for modified files
+        id: git-check
+        run: echo ::set-output name=modified::$(if git diff-index --quiet HEAD --; then echo "false"; else echo "true"; fi)
+      - name: Push changes
+        if: steps.git-check.outputs.modified == 'true'
+        run: |
+          git config --global user.email "AutoFmtBot@fmtyacodeffs.com"
```

Comment:
I did a quick whois on this address and it didn't bring anything up.  Is it just a dummy address that you came up with?

Reply:
Yep. It's a required field for doing an automatic push so I just made it up.

Reply:
this email is my favorite thing about this, as wonderful as it is